server/src/serverconfig.js

```diff
@@ -6,12 +6,17 @@
 import fs from 'fs'
 import path from 'path'
 
+// import.meta.dirname is undefined when using docker dev environment
+const url = import.meta.url
+const directoryURL = new URL('.', url)
+const directoryPath = directoryURL.pathname
```

Comment:
Should be simplified as `const directoryPath = import.meta.dirname || (new URL('.', import.meta.urll)).pathname`.

Also, not sure why `import.meta.dirname` is undefined in dev container, since it uses the same `ppbase` image that `ppserver` and `ppfull` image use, and `server/src/app.ts` works in those containers.  I think there is an underlying bug that is specific to the dev container, that should be addressed, instead of modifying `app.ts` and `serverconfig.js`.